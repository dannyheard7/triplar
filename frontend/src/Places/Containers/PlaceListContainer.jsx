import React from "react";
import PlaceListItem from "Places/Components/PlaceListItem";

import "Places/styles/places.css";
import {Draggable} from "react-beautiful-dnd";


export default class PlaceListContainer extends React.Component {
    render() {
        return (
            <div>
                {this.props.places.map((item, index) => (
                    <Draggable key={item.id} draggableId={item.id} index={index}>
                        {(provided, snapshot) => (
                            <div ref={provided.innerRef} {...provided.draggableProps} {...provided.dragHandleProps}>
                                <PlaceListItem key={item.id} place={item} path={this.props.path}/>
                            </div>
                        )}
                    </Draggable>
                ))}
            </div>
        );
    }
}
