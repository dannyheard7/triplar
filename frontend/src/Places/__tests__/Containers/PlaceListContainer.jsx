import React from "react";

import ShallowRenderer from 'react-test-renderer/shallow';
import PlaceListContainer from "Places/Containers/PlaceListContainer";

const renderer = new ShallowRenderer();

describe('<PlaceListContainer />', () => {
    test('renders correctly', () => {
        const places = [{'id': 1, 'name': 'Place 1', 'imageUrl': 'example.com/place1.jpg'},
                        {'id': 2, 'name': 'Place 2', 'imageUrl': 'example.com/place2.jpg'}];
        const result = renderer.render(<PlaceListContainer places={places} />);
        expect(result).toMatchSnapshot();
    });
});