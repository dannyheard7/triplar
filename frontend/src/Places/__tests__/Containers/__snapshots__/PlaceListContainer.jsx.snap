// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceListContainer /> renders correctly 1`] = `
<div>
  <Connect(Draggable)
    disableInteractiveElementBlocking={false}
    draggableId={1}
    index={0}
    isDragDisabled={false}
  >
    [Function]
  </Connect(Draggable)>
  <Connect(Draggable)
    disableInteractiveElementBlocking={false}
    draggableId={2}
    index={1}
    isDragDisabled={false}
  >
    [Function]
  </Connect(Draggable)>
</div>
`;
